addons/stock/models/stock_picking.py

```diff
@@ -417,17 +418,28 @@ def _compute_state(self):
         - Done: if the picking is done.
         - Cancelled: if the picking is cancelled
         '''
+        picking_moves_state_map = defaultdict(dict)
+        picking_move_lines = defaultdict(set)
+        for move in self.env['stock.move'].search([('picking_id', 'in', self.ids)]):
```

Comment:
equivalent to `self.mapped("move_lines")` ?
I don't get it why the `picking.move_lines` is slow, the prefetch should prefetch move_lines for self, no ? (Or maybe there isn't a prefetch for the one2many in 13.0 ? )

Reply:
The result is the same but `self.env['stock.move].search` is faster here

Reply:
Well I also thought that move_lines should be prefetched but looking at the flamegraph, I saw that there was a CacheMiss on move_lines triggering a call to `fetch_field`. So, since it loops through moves to compute the `picking_moves_state_map` anyways, it was faster to preconstruct moves sets.

Reply:
I tested the flow : 
- The move_lines is prefetch correctly but it is the `_in_cache_without` give a big extra cost when we read `move_lines` by picking. No sure if this issue exist in the version bigger than 13.0, also normaly in 14.0 the compute store is batched more efficiently (at the end of request), then your changes should be more effective in 14.0.
- I am not fan of using a search instead of the usage of `self.move_lines` (we lose the ORM cache of `move_lines`). But obviously, in this case, it is legit (?).

Then, ok with your solution (Maybe we should retest the performance gain in version > 13.0).